### PR TITLE
Reconnect to MQTT broker if connection is dropped

### PR DESCRIPTION
**Pull request by qwandor**
This should fix the `homie-influx` part of #7. It also makes errors writing to InfluxDB non-fatal. There is no point crashing just because one database isn't working, as the others may still work.

### CODE REVIEW

File: homie-influx/src/main.rs [hunk 1]
```diff
@@ -31,64 +32,86 @@ async fn main() -> Result<(), eyre::Report> {
...
+            influxdb_client,
+            config.mqtt.reconnect_interval,
+        );
         controller.start().await?;
-        join_handles.push(handle.map(|res| Ok(res??)));
+        join_handles.push(handle);
     }
 
-    simplify_unit_vec(try_join_all(join_handles).await)
+    simplify_unit_vec(try_join_all(join_handles).await)?;
```
Comment (alsuren) [inline homie-influx/src/main.rs:45]: No need to simplify it you are throwing the result away

File: homie-influx/src/config.rs [hunk 2]
```diff
@@ -3,15 +3,18 @@ use influx_db_client::reqwest::Url;
...
 use stable_eyre::eyre;
 use stable_eyre::eyre::WrapErr;
 use std::fs::read_to_string;
 use std::sync::Arc;
+use std::time::Duration;
 
 const DEFAULT_MQTT_CLIENT_PREFIX: &str = "homie-influx";
 const DEFAULT_MQTT_HOST: &str = "test.mosquitto.org";
 const DEFAULT_MQTT_PORT: u16 = 1883;
+const DEFAULT_MQTT_RECONNECT_INTERVAL: Duration = Duration::from_secs(5);
```
Comment (alsuren) [inline homie-influx/src/config.rs:17]: How long does it take to get synced up with all the mqtt state? If you were in a fresh loop, what fraction of your time would be spent hammering the server?

**Review by alsuren — APPROVED** [review]
Looks sensible. Hopefully it fixes your dad's problem.

File: homie-influx/src/config.rs (hunk 2 above)
Comment (qwandor) [reply homie-influx/src/config.rs:17]: The state is preserved on the broker when `clean_session` is set to false, so the client just needs to authenticate and connect/conack. WIthout a delay it just seems to constantly try to reconnect. I didn't check CPU usage, but there was certainly a lot of log spam.

File: homie-influx/src/main.rs (hunk 1 above)
Comment (qwandor) [reply homie-influx/src/main.rs]: Done.

**Pull request merged**